### ISSUE TITLE
clone(deep = TRUE) is not `all.equal` to original

### ISSUE THREAD

**Issue by mb706** [mlr-org/paradox#149]
```r
x = ParamSet$new()
all.equal(x$clone(deep = TRUE), x)
```
gives
```r
[1] "Component “deps”: Modes: list, NULL"       
[2] "Component “deps”: current is not list-like"
```
This is because `x$deps` is `NULL` while `x$clone(deep = TRUE)$deps` is `list()`.

**Comment by mb706** [comment]
This is breaking tests in mlr3pipeops

**Comment by berndbischl** [comment]
fixed by PR #150
